### ISSUE TITLE
A probolem for muti-timesteps input factors using sobol sensitivity analysis

### ISSUE THREAD

**Issue by BoltsLei** [SALib/SALib#503]
Hi, I have a model which can predict the water quality. But the input factors are all coninues variable(time series), such as y = _**f**_(_**x1, x2, x3**_) , and _**xi**_ has the _t_ time step input, and _i_ belongs to [1, k]. In the other word, the model's input is a (t * k) matix. I want to analyze the contribution for every component in all _t_ time. How I can do this?

My current idea is to treat every time_step of factors as a singe varaiable ,  and then sampe every varaiable independently .
Finally, group t time steps varaiable into a component or sum the every varaiable in time diemention.
My definition of `problem` is below:
```python
 problem = {
        'num_vars': TIME_STEPS * INPUT_SIZE,
        'names': [str(name) for name in range(TIME_STEPS * INPUT_SIZE)],
        'bounds': numpy.asarray(bounds_np),
        'groups': ['Group-' + str(idx % INPUT_SIZE) for idx in range(TIME_STEPS * INPUT_SIZE)],
    }
```
Is my approach feasible or reansonable? How I can count this time series input factors sensitivity?

**Comment by willu47** [comment]
Please take a look at @jdherman's paper ["From maps to movies: high-resolution time-varying sensitivity analysis for spatially distributed watershed models"](https://hess.copernicus.org/articles/17/5109/2013/)

**Comment by BoltsLei** [comment]
@willu47 Thank you very much for this paper. 
I have another question about HDMR method. My master study is about water quality prediction research.  I have many time series data about river (a point collected from auto monitor station)and pollution source (200 number, eg. company) lying by the river. Certain sources of pollution may affect river water quality (eg, monitored by certain elements). I would like to identify possible individuals in a list of pollution sources to build a more robust water quality prediction model.

Recently I find a paper about `HDMR` that can construct a computationl model directly from lab/field data.

Can I use the `HDMR` to analyze the sensitivity between pollution source and river by these time series? I search a lot of paper to verify this idea, but no answer. Could you give me some advise when you have time. Thank you.

**Comment by ConnectedSystems** [comment]
> Can I use the HDMR to analyze the sensitivity between pollution source and river by these time series?

I think this is true of any SA approach that does not require a specific sample scheme. For HDMR specifically though, you may run into several practical issues.

For one, HDMR seems to be a very computationally intensive approach. It can use a lot of memory, particularly for large number of dimensions. You can try and see what happens.

**Comment by BoltsLei** [comment]
> > Can I use the HDMR to analyze the sensitivity between pollution source and river by these time series?
> 
> I think this is true of any SA approach that does not require a specific sample scheme. For HDMR specifically though, you may run into several practical issues.
> 
> For one, HDMR seems to be a very computationally intensive approach. It can use a lot of memory, particularly for large number of dimensions. You can try and see what happens.

Yeah, HDMR methods can't resolve the problem with more than 100 parameters and it will consume lots of memory. In the last, I   group parameters into dimensions to reduce analyze time by using Morris or Sobol method.
